sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,428 @@
+
+    @Test
+    public abstract void detectLanguagesBatchStringInput();
+
+    @Test
+    public abstract void detectLanguagesBatchListCountryHint();
+
+    void detectLanguageShowStatisticsRunner(BiConsumer<List<DetectLanguageInput>,
+        TextAnalyticsRequestOptions> testRunner) {
+        final List<DetectLanguageInput> detectLanguageInputs = Arrays.asList(
+            new DetectLanguageInput("0", "This is written in English", "US"),
+            new DetectLanguageInput("1", "Este es un document escrito en Español."),
+            new DetectLanguageInput("2", "~@!~:)", "US")
+            // add error document => empty text
+        );
+
+        testRunner.accept(detectLanguageInputs, setTextAnalyticsRequestOptions());
+    }
+
+    void detectLanguageDuplicateIdRunner(BiConsumer<List<DetectLanguageInput>,
+        TextAnalyticsRequestOptions> testRunner) {
+        final List<DetectLanguageInput> detectLanguageInputs = Arrays.asList(
+            new DetectLanguageInput("0", "This is written in English", "US"),
+            new DetectLanguageInput("0", "Este es un document escrito en Español.")
+        );
+
+        testRunner.accept(detectLanguageInputs, setTextAnalyticsRequestOptions());
+    }
+
+    static void detectLanguagesCountryHintRunner(BiConsumer<List<String>, String> testRunner) {
+        final List<String> inputs = new ArrayList<>(Arrays.asList(
+            "This is written in English", "Este es un document escrito en Español.", "~@!~:)"));
+
+        testRunner.accept(inputs, "US");
+    }
+
+    static void detectLanguageStringInputRunner(Consumer<List<String>> testRunner) {
+        final List<String> inputs = new ArrayList<>(Arrays.asList(
+            "This is written in English", "Este es un document escrito en Español.", "~@!~:)"));
+
+        testRunner.accept(inputs);
+    }
+
+    static void detectLanguageRunner(Consumer<List<DetectLanguageInput>> testRunner) {
+        final List<DetectLanguageInput> detectLanguageInputs = Arrays.asList(
+            new DetectLanguageInput("0", "This is written in English", "US"),
+            new DetectLanguageInput("1", "Este es un document escrito en Español."),
+            new DetectLanguageInput("2", "~@!~:)", "US")
+        );
+
+        testRunner.accept(detectLanguageInputs);
+    }
+
+    static DocumentResultCollection<DetectLanguageResult> getExpectedBatchDetectedLanguages() {
+        DetectedLanguage detectedLanguage1 = new DetectedLanguage().setName("English").setIso6391Name("en")
+            .setScore(1.0);
+        DetectedLanguage detectedLanguage2 = new DetectedLanguage().setName("Spanish").setIso6391Name("es")
+            .setScore(1.0);
+        DetectedLanguage detectedLanguage3 = new DetectedLanguage().setName("(Unknown)").setIso6391Name("(Unknown)")
+            .setScore(0.0);
+        List<DetectedLanguage> detectedLanguageList1 = new ArrayList<>(Collections.singletonList(detectedLanguage1));
```

Comment:
Collections.singletonList already returns a list. Why are you wrapping it in another ArrayList?